### PR TITLE
New prop additions

### PR DESCRIPTION
**Pull request by stevenyuen**
**Description**
- **TabMenu** - Ability to set button text props
- **Button** - button hover color is darkened if the button is a light color
- **Form** - ability to set `highlightColor` (which is then used by all child `Form` elements)

### CODE REVIEW

File: src/Form/Checkbox/index.js [hunk 1]
```diff
@@ -18,11 +17,10 @@ const Checkbox = React.forwardRef(({
    label,
    ...props
 }, ref) => {
-   const oioContext = useContext(OIOContext)
    const formContext = useContext(OIOFormContext)
 
    const inputAppearance = appearance || formContext.elementAppearance
-   const inputHighlightColor = highlightColor || oioContext.highlightColor
+   const inputHighlightColor = highlightColor || formContext.highlightColor
```
Comment (asabhaney) [inline src/Form/Checkbox/index.js]: Just curious, how come you removed the `oioContext.highlightColor`, what if the `Checkbox` is not used in a form? Why not do:

```js
const inputHighlightColor = highlightColor || formContext.highlightColor || oioContext.highlightColor
```

File: src/Form/Radio/index.js [hunk 2]
```diff
@@ -17,10 +16,9 @@ const Radio = React.forwardRef(({
    label,
    ...props
 }, ref) => {
-   const oioContext = useContext(OIOContext)
    const formContext = useContext(OIOFormContext)
 
-   const inputHighlightColor = highlightColor || oioContext.highlightColor
+   const inputHighlightColor = highlightColor || formContext.highlightColor
```
Comment (asabhaney) [inline src/Form/Radio/index.js]: Same question about oio context

File: src/Form/Switch/index.js [hunk 3]
```diff
@@ -3,15 +3,16 @@ import PropTypes from 'prop-types'
...
 
 // ============================================================================
 // Component
 // ============================================================================
 
 const Switch = React.forwardRef(({ highlightColor, ...props }, ref) => {
-   const oioContext = useContext(OIOContext)
-   const inputHighlightColor = highlightColor || oioContext.highlightColor
+   const formContext = useContext(OIOFormContext)
+   const inputHighlightColor = highlightColor || formContext.highlightColor
```
Comment (asabhaney) [inline src/Form/Switch/index.js]: Same question about oio context

File: src/TabMenu/spec.js [hunk 4]
```diff
@@ -28,6 +28,30 @@ export default [{
    default: '<code>0px</code>',
    description: 'Set spacing between <code>TabMenuButton</code> components.',
    responsive: false
+}, {
+   name: 'buttonTextColor',
+   type: 'String',
+   default: '<code>#888</code>',
+   description: 'Text <code>color</code> of the <code>TabMenuButton</code> when <code>isActive</code> is set to <code>false</code>',
```
Comment (asabhaney) [inline src/TabMenu/spec.js]: What happens when it is active for these new props?

File: src/TabMenu/spec.js [hunk 5]
```diff
@@ -28,6 +28,30 @@ export default [{
...
+   name: 'buttonTextSize',
+   type: 'OIO Responsive String',
+   default: '-',
+   description: 'Set the size of the text inside <code>TabMenuButton</code> using a number between <code>1-15</code>. If no value is specified, text will be sized automatically based on the <code>TabMenuButton</code> <code>size</code>',
+   responsive: true
+}, {
+   name: 'buttonTextWeight',
+   type: 'OIO Responsive String',
+   default: '<code>medium</code>',
+   description: 'Set the <code>fontWeight</code> of the text inside <code>TabMenuButton</code> when <code>isActive</code> is set to <code>false</code>',
```
Comment (asabhaney) [inline src/TabMenu/spec.js]: Actually this seems to affect all buttons, including the active one

File: src/TabMenu/spec.js [hunk 6]
```diff
@@ -28,6 +28,30 @@ export default [{
...
+   description: 'Text <code>color</code> of the <code>TabMenuButton</code> when <code>isActive</code> is set to <code>false</code>',
+   responsive: false
+}, {
+   name: 'buttonTextFontFamily',
+   type: 'OIO Responsive String',
+   default: '-',
+   description: 'Set <code>fontFamily</code> for all <code>TabMenuButton</code> components.',
+   responsive: true
+}, {
+   name: 'buttonTextSize',
```
Comment (asabhaney) [inline src/TabMenu/spec.js]: If the size is a bit bigger, the text ends up being taller than the actual component, but there doesn't seem to be a way to adjust the height of the `TabMenu`?

Screen Shot 2019-11-09 at 4 54 09 PM

File: src/Button/index.js [hunk 7]
```diff
@@ -125,7 +125,9 @@ const pulsingAnimation = keyframes`
       borderStyle: 'none',
       borderWidth: 'none',
       textSize: textSize || buttonSizeDefaults[size].textSize,
-      hoverBackgroundColor: tinycolor(primaryButtonColor).lighten(7).toString(),
+      hoverBackgroundColor: tinycolor(primaryButtonColor).isDark()
```
Comment (asabhaney) [inline src/Button/index.js]: Cool!

File: src/TabMenu/spec.js (hunk 6 above)
Comment (stevenyuen) [reply src/TabMenu/spec.js]: Yeah, I assume that we will add a way to adjust the button's height manually, but I am still thinking about the best way to do so. There is a possibility that the `buttonSize` prop will be removed altogether, but I am still thinking about it.

File: src/TabMenu/spec.js (hunk 4 above)
Comment (stevenyuen) [reply src/TabMenu/spec.js]: The button's text when active is the `highlightColor`. I'm still figuring out these conventions to be honest - still seeing what the customization needs are in real-world usage.

Comment (stevenyuen) [reply src/TabMenu/spec.js]: It is possible that we will make it like `ListMenu` which also has a `activeTextColor` prop

**Pull request merged**